memory/src/columns.rs

```diff
@@ -27,31 +28,29 @@ pub struct MemoryCols<T> {
 
     /// A boolean flag indicating whether addr' - addr == 0
     pub addr_not_equal: T,
+
+    pub counter: T,
+}
+
+#[derive(AlignedBorrow, Default)]
+pub struct MemoryPermutationCols<T> {
```

Comment:
Is this permuted in the sense of being sorted by (addr, timestamp)? I think the original data in `MemoryCols` can be sorted like that, since the CPU table will have the chronological log.

Reply:
`MemoryCols` is sorted in that order: https://github.com/delendum-xyz/valida/blob/9902f119c4d759b94124b4f878ff92e8217f6457/memory/src/lib.rs#L98-L99

`MemoryPermutationCols` stores columns that need either sorting or permutations necessary for the range check. The `addr` column in that struct is already sorted correctly, so maybe we can remove it from this struct and use the one in `MemoryCols`. The `clk` column is not.